wye/regions/forms.py

```diff
@@ -28,6 +28,21 @@ def clean(self):
 
     def save(self, force_insert=False, force_update=False, commit=True):
         m = super(RegionalLeadForm, self).save()
+
+        # Removing old leads which are not selected currently
+        current_region = self.instance.location.id
+        lead_users = Profile.objects.filter(usertype=UserType.objects.get(slug='lead'))
+        current_region_leads = lead_users.filter(location=current_region)
+        current_leads = self.instance.leads.values()
```

Comment:
functionality is fine but we can reduce DB hit and for loops. Can you revisit code and look on optimizing it.